### ISSUE TITLE
Failed to query list identity. Can not perform fingerprinting.

### ISSUE THREAD

**Issue by clint70** [claroty/enip-stack-detector#3]
I'm hoping that I can get some clarity on why this is happening...  I have spent a fair amount of time on this, but have been unsuccessful in getting the script to run successfully.  It does seem to run... I have ran it through pdb and followed... seems that the data is a byte string that is empty?

pkt_list_identity: b'c\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
pkt_resp
b'c\x00\x00\x00\x00\x00\x00\x00\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'

Now, I am not sure, but I have tested the tcp socket similar to what is in the code here... it will connect, but I am thinking that what is happening is because there never was a req. for a session, the PLC won't reply??  BUT I am not sure.

the machine running the enip-stack-detector is on the same network, there are no firewall rules interfering with port 44818, and I can see the traffic.  The PLC is an Allen-Bradley 1756-A7 with several modules, and there is another 1756-B7, with the same or similar modules.  I have been successful with CIP connecting to both.  I can see the ENETB module and both are up.  Perhaps... this setup is not compatible?  BUT my confusion is with the fact that CIP / Ethernet/IP is supposed to be universal...

Would be really appreciative, if someone could provide me some insight, as I would like to understand what is going on.  Thanks for any help / support.

**Comment by SharonBrizinov** [comment]
Hi, could you share a pcap? it will be easier to understand what's going on from there, if not then maybe screenshot of the packets from Wireshark.

Also, can you try to run nmap with enip nse script and share if it works or not?
https://svn.nmap.org/nmap/scripts/enip-info.nse

**Comment by clint70** [comment]
I appreciate the reply... thank you.  Sorry I didn't provide better information.  As for sharing the pcap / other info.  Right now I am on a small training network, that is not particularly complex.  BUT I'll check if I can share. 

I used the the script, that worked.  I am a bit puzzled, port 44818 is def. open and connecting to my attempts to do so.  But the script did not find it?

I did run this and got something diff. 

```
└─$ nmap -n -sV --script enip-info -p 44818 10.10.25.90
Starting Nmap 7.93 ( https://nmap.org ) at 2023-02-24 08:20 EST
Nmap scan report for 10.10.25.90
Host is up (0.0055s latency).

PORT      STATE SERVICE       VERSION
44818/tcp open  EtherNetIP-2?
| fingerprint-strings: 
|   TLSSessionReq: 
|_    random1ra
1 service unrecognized despite returning data. If you know the service/version, please submit the following fingerprint at https://nmap.org/cgi-bin/submit.cgi?new-service :
SF-Port44818-TCP:V=7.93%I=7%D=2/24%Time=63F8B9C7%P=x86_64-pc-linux-gnu%r(S
SF:SLSessionReq,18,"\x16\x03\0\0S\x01\0\0\x01\0\0\0G\xd7\xf7\xba,\xee\xea\
SF:xb2`~\xf3\0")%r(TLSSessionReq,18,"\x16\x03\0\0i\x01\0\0\x01\0\0\0\x1c\x
SF:a7\xe4random1ra")%r(LDAPSearchReq,18,"0\x84\0\0\0-\x02\x01\x01\0\0\0\0\
SF:0\$\x04\0\n\x01\0\n\x01\0\x02")%r(NotesRPC,18,":\0\0\0/\0\0\0\x01\0\0\0
SF:\x02\x0f\0\x01\0=\x05\0\0\0\0\0")%r(oracle-tns,18,"\0Z\0\0\x01\0\0\0\x0
SF:1\0\0\0\0\0\x08\0\x7f\xff\x7f\x08\0\0\0\x01");

Service detection performed. Please report any incorrect results at https://nmap.org/submit/ .
Nmap done: 1 IP address (1 host up) scanned in 158.15 seconds
```

**Comment by clint70** [comment]
THANK YOU for the help and nmap pointer.  I am still not totally sure why 44818 didn't show up when I tried the script you linked me to, but I tried a diff. script found the port... and as above was able to fingerprint.  IN the end, I still don't know what happened, but diff workstation, reinstalled a virtualenv with python2.7 and the other requirements .... 

It worked.  Maybe something wrong with the prev. workstation.  Appreciate the time... and this is great work. 

Cheers!
